datafusion/expr/src/built_in_function.rs

```diff
@@ -571,6 +574,7 @@ impl BuiltinScalarFunction {
             BuiltinScalarFunction::ArrayDims => {
                 Ok(List(Arc::new(Field::new("item", UInt64, true))))
             }
+            BuiltinScalarFunction::ArrayDistinct => Ok(input_expr_types[0].clone()),
```

Comment:
Is it possible to add additional input check here like
```rust
match input_expr_types[0] {
    List(_) => Ok(input_expr_types[0].clone()),
    _ => return plan_err!("...")
}
```
To make this query an error instead of panic
```
❯ select array_distinct(1);
Optimizer rule 'simplify_expressions' failed
caused by
Internal error: could not cast value to arrow_array::array::list_array::GenericListArray<i32>.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
```

Reply:
return [] for array_distinct([])